experimental/Matrix/matrix.g

```diff
@@ -0,0 +1,139 @@
+  "IsJuliaMatrixRep",
+  IsComponentObjectRep and IsAttributeStoringRep and IsMatrixObj);
+  
+JuliaMatrixIdentificationType :=
+    NewType( JuliaMatrixFamily, IsJuliaMatrixRep);
+    
+#DeclareOperation( "MakeJuliaMatrixRep", [ IsJuliaObject] );
+
+#InstallMethod( MakeJuliaMatrixRep, "", [IsJuliaObject],
+#    function(m)
+#        return Objectify(JuliaMatrixIdentificationType, rec(m := m));
+#    end
+#);
+
+BindGlobal("MakeJuliaMatrixRep",function(m)
+  return Objectify(JuliaMatrixIdentificationType,
+               rec(m := m));
+end);
+
+#DeclareOperation( "\*", [ IsJuliaMatrixRep, IsJuliaMatrixRep ] );
+#DeclareOperation( "\/", [ IsJuliaMatrixRep, IsJuliaMatrixRep ] );
+
+#InstallMethod( \*,
+#    "for two IsJuliaMatrixRep",
+#    [ IsJuliaMatrixRep, IsJuliaMatrixRep ],
+#    function( m1, m2 )
+#        return MakeJuliaMatrixRep(m1!.m * m2!.m);
+#    end );
+
+#InstallMethod( \/,
+#    "for two IsJuliaMatrixRep",
+#    [ IsJuliaMatrixRep, IsJuliaMatrixRep ],
+#    function( m1, m2 )
+#        return m1 * Inverse(m2);
+#    end );
+    
+InstallOtherMethod( NumberRows, [IsJuliaMatrixRep], m -> Julia.nrows(m!.m));
+
+InstallOtherMethod( NumberColumns, [IsJuliaMatrixRep], m -> Julia.ncols(m!.m));
+
+InstallOtherMethod( BaseDomain, [IsJuliaMatrixRep], m -> Julia.base_ring(m!.m));
+
+InstallOtherMethod( \[\,\],
+    [ IsJuliaMatrixRep, IsPosInt and IsSmallIntRep,
+                       IsPosInt and IsSmallIntRep ],
+    function( m, i, j )
+      return (m!.m)[i,j];
+    end );
+    
+InstallOtherMethod( \*, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return MakeJuliaMatrixRep(m1!.m * m2!.m);
+    end );
+    
+InstallOtherMethod( \/, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1 * Inverse(m2);
+    end);
+    
+InstallMethod( InverseMutable, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(InverseOp(m!.m)));
+
+InstallMethod( InverseImmutable, [IsJuliaMatrixRep], m -> MakeImmutable(MakeJuliaMatrixRep(InverseOp(m!.m))));
+
+InstallOtherMethod( AdditiveInverseOp, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(AdditiveInverseOp(m!.m)));
+
+InstallOtherMethod(One, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(OneOp(m!.m)));
+
+InstallOtherMethod(Zero, [IsJuliaMatrixRep], m -> MakeJuliaMatrixRep(ZeroOp(m!.m)));
+
+InstallOtherMethod(\+, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return MakeJuliaMatrixRep(m1!.m + m2!.m);
+    end);
+    
+InstallOtherMethod(\-, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return MakeJuliaMatrixRep(m1!.m - m2!.m);
+    end);
+
+InstallOtherMethod(\^, [IsJuliaMatrixRep, IsInt and IsSmallIntRep], function( m1, k )
+        if k < 0 then
+            return MakeJuliaMatrixRep( (Inverse(m1)!.m)^(-k) );
+        else
+            return MakeJuliaMatrixRep( (m1!.m)^k );
+        fi;
+    end);
+    
+InstallOtherMethod(\=, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1!.m = m2!.m;
+    end);
```

Comment:
You'll also need a `\<` method

Reply:
I have one but I think that it needs more checks. For example the following error occurs:

''
julia> m < m3
ERROR: MethodError: no method matching isless(::fmpz_mat, ::fmpq_mat)
Closest candidates are:
  isless(::Any, ::Hecke.EmbeddedNumFieldElem) at
''

I.e. In some cases < is not defined for Julia matrices. What do you think?